### PR TITLE
Flink: Add an optional uidPrefix to FlinkSink#Builder to explicitly set operator uid.

### PR DESCRIPTION
**Pull request by stevenzwu**
…ator id.

### CODE REVIEW

**Comment by stevenzwu** [comment]
@openinx can you take a look? 

This is to address this email thread from mailing list: https://www.mail-archive.com/dev@iceberg.apache.org/msg02269.html

**Review by openinx — APPROVED** [review]

File: flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java [hunk 1]
```diff
@@ -205,6 +207,30 @@ public Builder equalityFieldColumns(List<String> columns) {
...
+     * pipeline.auto-generate-uids=false} to disable auto-generation and force explicit setting of all operator uids.
+     * <p>
+     * Be careful with setting this for an existing job, because now we are changing the opeartor uid from an
+     * auto-generated one to this new value. When deploying the change with a checkpoint, Flink won't be able to restore
+     * the previous Flink sink operator state (more specifically the committer operator state). You need to use {@code
+     * --allowNonRestoredState} to ignore the previous sink state. During restore Flink sink state is used to check if
+     * checkpointed files were actually committed or not. {@code --allowNonRestoredState} can lead to data loss if the
+     * Iceberg commit failed in the last completed checkpoints.
+     *
+     * @param newPrefix defines the iceberg table's key.
```
Comment (openinx) [inline flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java:226]: > @param newPrefix defines the iceberg table's key.

I think we will need a correct parameter doc for this.

**Comment by openinx** [comment]
@stevenzwu This PR looks pretty good to me, just left a small comment to fix this parameter comment !   The javadoc looks very friendly to datastream users.

**Comment by openinx** [comment]
Got this PR merged,  thanks @stevenzwu for contributing !

**Pull request merged**